spec/jobs/reports/combined_invoice_supplement_report_v2_spec.rb

```diff
@@ -8,341 +8,561 @@
+
+        let(:user8) { create(:user, profiles: [profile8]) }
+        let(:profile8) { build(:profile, verified_at: DateTime.new(2017, 1, 1).utc) }
+
+        let(:user9) { create(:user, profiles: [profile9]) }
+        let(:profile9) { build(:profile, verified_at: DateTime.new(2016, 1, 1).utc) }
+
+        let(:user10) { create(:user, profiles: [profile10]) }
+        let(:profile10) { build(:profile, verified_at: DateTime.new(2015, 1, 1).utc) }
+
+        let(:csv) { CSV.parse(report.perform(Time.zone.today), headers: true) }
+
+        before do
+          iaa_order2.integrations << build_integration(
+            issuer: iaa2_sp1.issuer,
+            partner_account: partner_account2,
+          )
+          iaa_order2.integrations << build_integration(
             issuer: iaa2_sp2.issuer,
-            requested_at: inside_iaa2,
-            returned_at: inside_iaa2,
-            profile_verified_at: DateTime.new(2019, 1, 1).utc,
-            billable: true,
+            partner_account: partner_account2,
+          )
+          iaa_order2.save
+
+          # ----- iaa2_sp1 sp_return_logs -----
+          # 1 new unique user in month 1 at IAA 2 sp 1 @ IAL 2 with profile age 1
+          create_sp_return_log(
+            user: user1, issuer: iaa2_sp1.issuer, ial: 2,
+            returned_at: inside_iaa2
+          )
+
+          # 1 new unique user in month 1 at IAA 2 sp 1 @ IAL 2 with profile age 3
+          create_sp_return_log(
+            user: user7, issuer: iaa2_sp1.issuer, ial: 2,
+            returned_at: inside_iaa2
+          )
+
+          # 1 new unique user in month 1 at IAA 2 sp 1 @ IAL 2 with profile age 5
+          create_sp_return_log(
+            user: user9, issuer: iaa2_sp1.issuer, ial: 2,
+            returned_at: inside_iaa2
+          )
+
+          # 1 new unique user in month 1 at IAA 2 sp 1 @ IAL 2 with profile age unknown
+          create_sp_return_log(
+            user: user4, issuer: iaa2_sp1.issuer, ial: 2,
+            returned_at: inside_iaa2
+          )
+
+          # ----- iaa2_sp2 sp_return_logs -----
+          # 2 new unique user in month 1 at IAA 2 sp 2 @ IAL 2 with profile age 2
+          [user5, user6].each do |user|
+            create_sp_return_log(
+              user: user, issuer: iaa2_sp2.issuer, ial: 2,
+              returned_at: inside_iaa2
+            )
+          end
+
+          # 1 new unique user in month 1 at IAA 2 sp 2 @ IAL 2 with profile age 4
+          create_sp_return_log(
+            user: user8, issuer: iaa2_sp2.issuer, ial: 2,
+            returned_at: inside_iaa2
           )
+
+          # 1 new unique user in month 1 at IAA 2 sp 2 @ IAL 2 with profile age > 5
+          create_sp_return_log(
+            user: user10, issuer: iaa2_sp2.issuer, ial: 2,
+            returned_at: inside_iaa2
+          )
+        end
+
+        it 'validates all profile age columns and mulitple issuers for single partner' do
```

Comment:
again, "validates" is vague, also typo in "multiple"
```suggestion
        it 'has correct values for all profile age columns and multiple issuers for single partner' do
```